### DISCUSSION TITLE
大型脚本执行时出现工单状态异常

### DISCUSSION THREAD

**Discussion by peixubin** [hhyo/Archery#2651] [Q&A]
archery版本为最新版，docker image sha：sha256:2187bb2c116ce9af21509946a625e4e9c299ed448363d059a2c9bb418f01fc8d。

在执行小型脚本时，没有错误。
但是在执行一个4M长度的脚本时，出现以下错误：
`    
工单状态不正确，禁止执行！ : Traceback (most recent call last):
File "/opt/venv4archery/lib/python3.11/site-packages/django_q/cluster.py", line 432, in worker
res = f(*task["args"], **task["kwargs"])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/opt/archery/sql/utils/execute_sql.py", line 25, in execute
raise Exception("工单状态不正确，禁止执行！")
Exception: 工单状态不正确，禁止执行！

`
如何排查原因？

**Comment by LeoQuote** [comment]
简单搜索就可以搜到这个报错信息 https://github.com/hhyo/Archery/blob/14fbd5ed22f63541258c89b5223666a4fc7b1dfa/sql/utils/execute_sql.py#L25

可以多加一些日志看看具体情况

**Reply by LeoQuote** [reply]
我猜测是内部可能有某种重试，导致代码重复执行了，这可能是某种保护机制，你可以看看有没有任务异常退出的情况，比如oom 之类的

**Comment by peixubin** [comment]
看上去是默认的retry参数和timeout太短了

**Reply by LeoQuote** [reply]
那可以改一下 django q的参数哈
